### PR TITLE
KeyError#receiver and KeyError#key

### PR DESCRIPTION
**Pull request by ksss**
This is an implementation for [[Feature #12063]](https://bugs.ruby-lang.org/issues/12063)

### CODE REVIEW

**Comment by znz** [comment]
Matz approved this feature, but this PR has conflicts now.
@ksss Could you resolve conflicts?

**Comment by ksss** [comment]
@znz Thank you for the notice.
I rebased branch.

File: NEWS [hunk 1]
```diff
@@ -10,6 +10,11 @@ reference information is supplied with.  For a full list of changes
 with all sufficient information, see the ChangeLog file or Redmine
 (e.g. <tt>https://bugs.ruby-lang.org/issues/$FEATURE_OR_BUG_NUMBER</tt>)
 
+* KeyError
```
Comment (nobu) [inline NEWS]: This seems to be placed at "Core classes updates".

Comment (ksss) [inline NEWS]: Thanks ✨

File: NEWS [hunk 2]
```diff
@@ -10,6 +10,13 @@ reference information is supplied with.  For a full list of changes
 with all sufficient information, see the ChangeLog file or Redmine
 (e.g. <tt>https://bugs.ruby-lang.org/issues/$FEATURE_OR_BUG_NUMBER</tt>)
 
+=== Core classes updates (outstanding ones only)
```
Comment (nobu) [inline NEWS]: It is at the line 21 already.

Comment (ksss) [inline NEWS]: Thanks.
I misunderstood `since` and `ago`.

**Comment by hsbt** [comment]
Applied at https://github.com/ruby/ruby/commit/13f5dcb9f25cf07ef22baa5aded490395700b283

**Pull request closed without merging**